coderd/workspaceapps/apptest/apptest.go

```diff
@@ -0,0 +1,1151 @@
+package apptest
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/http/cookiejar"
+	"net/http/httputil"
+	"net/url"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/coderd/coderdtest"
+	"github.com/coder/coder/coderd/rbac"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/testutil"
+)
+
+// RunTests runs the entire workspace app test suite against deployments minted
+// by the provided factory.
+func RunTests(t *testing.T, factory DeploymentFactory) {
```

Comment:
Because this is _just_ used in `workspaceapps_test.go` and isn't tested independently, should it be it's own package?

I know it's large in size, but I might argue that it should be in the `workspaceapps_test` package instead, because that's what it's being used for anyways.

Regardless, we could call this `Run` instead, since `test` is in the package name.

Reply:
I think we can move it to `workspaceapps` pkg too.

We will call this from the APGL and enterprise coderd tests.